### PR TITLE
[Backend] Project Team Endpoints + Tests!

### PR DESCRIPTION
**Pull request by jzeiders**
- Adds the Project Team Endpoints
- Sets up Jest & Supertest for API Testing

TODO:
- [x] Finish Endpoints
- [ ] Plug into CLI

Note: already found a bug in the Person API when setting up Tests.

### CODE REVIEW

**Comment by saresend** [comment]
Request to have this on CI once you're done

File: api/database/database.js [hunk 1]
```diff
@@ -4,13 +4,15 @@ module.exports = {
     password: process.env.DEV_DB_PASSWORD,
     database: process.env.DEV_DB_NAME,
     host: process.env.DEV_DB_HOSTNAME,
+    logging: false,
```
Comment (saresend) [inline api/database/database.js]: Is this a sequelize specific thing?

File: api/people.js [hunk 2]
```diff
@@ -9,14 +9,14 @@ router.use(utils.authMiddleware);
 
 router.get("/self", async (req, res) => {
   try {
-    const pointsProfile = await models.Person.findOrCreate({
+    const [person, isCreated] = await models.Person.findOrCreate({
```
Comment (saresend) [inline api/people.js]: Where do we reference `isCreated`?

File: api/projectTeam.js [hunk 3]
```diff
@@ -8,21 +8,36 @@ router.use(utils.preprocessRequest);
...
 router.get("/self", async (req, res) => {
   try {
-    const { id } = req.user.id;
+    const { id } = req.user;
 
-    const person = await models.Person.findById(id);
+    const person = await models.Person.findByPk(id);
 
-    const projectTeam = await models.ProjectTeam.findById(person.teamId);
+    const projectTeam = await models.ProjectTeam.findByPk(person.ProjectTeamId);
```
Comment (saresend) [inline api/projectTeam.js]: Can Save network by using one query instead: 

```javascript
const personProjectTeam = await models.Person.findOne({
    where: {
       identityId: id
    }, 
    includes: [{ model: models.ProjectTeam, required: true}]
});
```

File: api/projectTeam.js [hunk 4]
```diff
@@ -55,15 +66,16 @@ router.post("/self", async (req, res) => {
...
 
     const result = await models.sequelize.transaction(async t => {
-      const projectTeam = await models.ProjectTeam.create(body, {
+      projectTeam = await models.ProjectTeam.create(body, {
         transaction: t
       });
 
       await person.setProjectTeam(projectTeam, { transaction: t });
     });
-    return res.json({ projectTeam: result });
```
Comment (saresend) [inline api/projectTeam.js]: Why are we using transactions here?

**Comment by saresend** [comment]
@jzeiders Can you explain why we're refactoring super `server.js` super heavily? Not necessarily against it just wanna understand why

File: api/database/database.js (hunk 1 above)
Comment (jzeiders) [reply api/database/database.js]: yeah, it's so it doesn't print every single request

File: api/people.js (hunk 2 above)
Comment (jzeiders) [reply api/people.js]: we don't but findorcreate returns a pair that needs to be deconstructed.

File: api/projectTeam.js (hunk 3 above)
Comment (jzeiders) [reply api/projectTeam.js]: Got me at my own game

File: api/database/database.js (hunk 1 above)
Comment (saresend) [reply api/database/database.js]: Oh, can we keep it in there so we can pipe logs eventually

File: api/projectTeam.js (hunk 4 above)
Comment (jzeiders) [reply api/projectTeam.js]: Create and assign ProjectTeam, don't want an orphaned teams

File: api/people.js (hunk 2 above)
Comment (saresend) [reply api/people.js]: This is news to me

Comment (jzeiders) [reply api/people.js]: Yeah we were returning the wrong value

**Comment by jzeiders** [comment]
Supertest can handle running and tearing down of our server for us. But Sever.js both handled the creation and running of the server. Additionally, this lets us test the express sever without spinning up next.js which saves several seconds on each re-compile.

File: api/database/database.js (hunk 1 above)
Comment (jzeiders) [reply api/database/database.js]: I left in for prod

**Pull request merged**